### PR TITLE
APNG Support/2.6.0

### PR DESCRIPTION
**Pull request by Iamgoofball**
**Takes advantage of QtApng to save everyone time and headache of rolling our own apng solution.**
https://github.com/Skycoder42/QtApng

You'll want to recompile it as a static plugin and do all the needed crap to get that to work along with the patched deps it uses with your setup, your choice as long as it compiles t b h

Beta compile of this in the mods channel and handed out to apng people so they can get cool shots of it working in game with new content to show why it's the superior format or something

Thanks to the Danganronpa Online dev(s) and Longbyte for assisting in getting the actual qt side of things working.

### CODE REVIEW

File: text_file_functions.cpp [hunk 1]
```diff
@@ -360,6 +360,22 @@ QString AOApplication::get_sfx_suffix(QString sound_to_check)
...
+    QString apng_check = path_to_check + ".apng";
+    QString gif_check = path_to_check + ".gif";
+    if(file_exists(apng_check))
+    {
+        return path_to_check + ".apng";
+    }
+    if(file_exists(gif_check))
+    {
+        return path_to_check + ".gif";
+    }
```
Comment (oldmud0) [inline text_file_functions.cpp]: Unnecessary branch.

File: courtroom.cpp [hunk 2]
```diff
@@ -703,7 +703,7 @@ void Courtroom::enter_courtroom(int p_cid)
   QString char_path = ao_app->get_character_path(current_char);
 
   if (ao_app->custom_objection_enabled &&
-      file_exists(char_path + "custom.gif") &&
+      (file_exists(char_path + "custom.gif") || file_exists(char_path + "custom.apng")) &&
       file_exists(char_path + "custom.wav"))
```
Comment (oldmud0) [inline courtroom.cpp:707]: You might want to take a look at changing this line if you want to pave support for Opus interjections.

File: aocharmovie.cpp [hunk 3]
```diff
@@ -3,6 +3,8 @@
 #include "misc_functions.h"
 #include "file_functions.h"
 #include "aoapplication.h"
+#include "debug_functions.h"
+#include <string>
```
Comment (oldmud0) [inline aocharmovie.cpp]: What's the point of this `<string>` include? Doesn't look like it's ever used.

**Review by oldmud0 — APPROVED** [review]
Looks very good - it's just about the same set of changes I would have made.

File: courtroom.cpp (hunk 2 above)
Comment (Iamgoofball) [reply courtroom.cpp:707]: Was going to do that in a follow up PR, to keep thing atomized.

File: aocharmovie.cpp (hunk 3 above)
Comment (Iamgoofball) [reply aocharmovie.cpp]: Leftover, same w/ the debug functions call. Will remove.

**Comment by Iamgoofball** [comment]
Addressed the issues in the review.

**Pull request merged**